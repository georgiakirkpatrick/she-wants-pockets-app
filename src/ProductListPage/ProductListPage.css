* {
    box-sizing: border-box;
}

.ProductListPage {
    padding: 0 15px;
}

.ProductListPage__product-list {
    /* list-style-type: none; */
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    margin: 1em 0 0;
    margin-block-start: 0;
    margin-block-end: 0;
    padding-inline-start: 0;
}