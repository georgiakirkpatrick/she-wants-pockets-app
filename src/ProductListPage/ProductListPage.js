import React from 'react'
import Product from '../Product/Product'
import FilterButtonSection from '../FilterButtonSection/FilterButtonSection'
import './ProductListPage.css'


function ProductListPage(props) {
    const generateProductList = props.products.map(product => (
        <Product 
            key={product.id} 
            id={product.id} 
            pathToImg={product.pathToImg} 
            imgAlt={product.imgAlt} 
            brand={product.brand} 
            productTitle={product.productTitle} 
            price={product.price}
        />
    ))
    
    return (
        <section className='ProductListPage'>
            <header>
                <h1>
                    Dresses
                </h1>
            </header>

            <FilterButtonSection selectedFilters={props.selectedFilters} />
            
            <ul className='ProductListPage__product-list'>
                {generateProductList}
            </ul>
        </section>
    )
}

ProductListPage.defaultProps = {
    products: []
}

export default ProductListPage