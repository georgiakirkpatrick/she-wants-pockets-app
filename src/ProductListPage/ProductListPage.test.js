import React from 'react';
import ReactDOM from 'react-dom';
import ProductListPage from './ProductListPage';
import renderer from 'react-test-renderer';

const productList = [
    {
        id: 1619,
        category: 'Dresses',
        categoryId: 'dresses',
        pathToImg: 'www.google.com',
        imgAlt: 'A dress',
        brand: 'Sezane',
        productTitle: 'The Georgia Dress',
        price: '98'
    }
]

describe ('ProductListPage', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ProductListPage />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ProductListPage productList={productList} />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})