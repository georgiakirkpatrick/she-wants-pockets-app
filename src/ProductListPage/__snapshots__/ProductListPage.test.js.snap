// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPage renders the UI as expected 1`] = `
<section
  className="ProductListPage"
>
  <header>
    <h1>
      Dresses
    </h1>
  </header>
  <div
    className="FilterButtonSection"
  >
    <div
      className="FilterButtonSection__button-section"
    >
      <button>
        Filter
      </button>
      <button>
        Sort
      </button>
    </div>
    <ul
      className="FilterButtonSection__tag-section"
    />
  </div>
  <ul
    className="ProductListPage__product-list"
  />
</section>
`;
